src/main/java/com/commilitio/medicalclinic/mapper/FacilityMapper.java

```diff
@@ -1,12 +1,33 @@
 package com.commilitio.medicalclinic.mapper;
 
+import com.commilitio.medicalclinic.model.Doctor;
 import com.commilitio.medicalclinic.model.Facility;
 import com.commilitio.medicalclinic.model.FacilityDto;
 import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 @Mapper(componentModel = "spring")
 public interface FacilityMapper {
+
+    @Mapping(target = "doctors", qualifiedByName = "mapToDoctor")
     FacilityDto toDto(Facility facility);
+
+    @Mapping(target = "doctors", qualifiedByName = "mapToDoctor")
     List<FacilityDto> toDtos(List<Facility> facilities);
+
+    @Named("mapToDoctor")
+    default Set<Long> mapDoctors(Set<Doctor> doctors) {
+        if (doctors == null) {
```

Comment:
return Optional.ofNullable(doctors)
.map( doctors -> doctors.stream().map(Doctor::getId).collect(Collectors.toSet()))
.orElse(new HashSet<>());

Obudowujemy w optionala doctors czyli jak doctors == null to sie nie wykonuje map tylko od razu orElse
Natomiast jak jest dostarczony to wykonuje sie map potem orElse